### PR TITLE
DAS-1901: Adding publishVariableDrafts field to publish generated variable drafts.

### PR DESCRIPTION
**Pull request by william-valencia**
# Overview

### What is the feature?

DAS-1901: Adding the publishVariableDrafts field to the collection query to use earthdata-varinfo to generate and publish variable drafts.

### What is the Solution?

Added publishVariableDrafts field to the collection query to use earthdata-varinfo to generate and publish variable drafts. Also obtained the rest of the variable info from CMR since earthdata-varinfo only returns the concept ids.

### What areas of the application does this impact?

the collection Query will have an additional field for publishVariableDrafts

# Testing

### Reproduction steps

- Need to test in UAT or PROD
- C1245618475-EEDTEST or similar

1. Use Apollo server to query collection publishVariableDrafts field. See README.

### Attachments

Please include relevant screenshots or files that would be helpful in reviewing and verifying this change.

# Checklist

- [ ] I have added automated tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings

### CODE REVIEW

**Review by macrouch — COMMENTED** [review]
publishVariableDrafts needs to be a mutation, not a return field within a query. 

Other work being done this PI also makes the name publishVariableDrafts too similar to the publishDraft mutation, so maybe something like publishGeneratedVariables would be more clear about what exactly is expected of the mutation

File: src/datasources/collectionVariableDrafts.js [hunk 1]
```diff
@@ -16,11 +16,7 @@ import { parseError } from '../utils/parseError'
 export default async (params, context) => {
   const { headers } = context
 
-  const { authorization: authorizationHeader = '' } = downcaseKeys(headers)
-
-  // Split the token by a space (Bearer xxx.xxx.xxx) and take the
-  // last element to get just the token without the 'Bearer' identifier
-  const token = authorizationHeader.split(' ').pop()
+  const { authorization: token } = downcaseKeys(headers)
```
[the remark this answers is not in the document]
Comment (william-valencia) [reply src/datasources/collectionVariableDrafts.js:19]: earthdata-varinfo  version 2.0.0(and up) wants the authorization header now, not just the token.

File: src/types/variable.graphql [hunk 2]
```diff
@@ -69,6 +69,11 @@ type VariableList {
   items: [Variable]
 }
 
+type PublishedVariableList {
```
[the remark this answers is not in the document]
Comment (william-valencia) [reply src/types/variable.graphql:72]: I am good with renaming it to publishGeneratedDrafts. We were going back and forth a bit on whether to make it a mutation vs a Query. We wanted to leverage the collection search ability to the collection Query to get the proper collection to use.

[the remark this answers is not in the document]
Comment (william-valencia) [reply src/types/variable.graphql]: Converted to mutation

File: src/datasources/collectionVariableDrafts.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (william-valencia) [reply src/datasources/collectionVariableDrafts.js]: renamed to authHeader

**Review by abbottry — APPROVED** [review]

File: src/resolvers/variable.js [hunk 3]
```diff
@@ -4,6 +4,39 @@ import { handlePagingParams } from '../utils/handlePagingParams'
...
+      const { dataSources } = context
+
+      const results = await dataSources.collectionVariableDraftsSource({
+        conceptId,
+        publish: true
+      }, context, parseResolveInfo)
+
+      // Pull out the variable concept ids from the source to use as parameters later
+      const conceptIds = {
+        params: { conceptId: Object.values(results).map((object) => Object.values(object)).flat() }
```
Comment (eudoroolivares2016) [inline src/resolvers/variable.js:20]: Can we split this up a bit or double check these all calls need to be made here?

File: src/earthdataVarinfo/test/expected_publish_response.json [hunk 4]
```diff
@@ -0,0 +1,9 @@
+{
+    "isBase64Encoded": false,
+    "statusCode": 200,
+    "body": [
+        {"conceptId": "V0001-TEST"},
+        {"conceptId": "V0002-TEST"}
+    ]
+}
+
```
Comment (eudoroolivares2016) [inline src/earthdataVarinfo/test/expected_publish_response.json:9]: Add newline

File: src/earthdataVarinfo/test/variables.json [hunk 5]
```diff
@@ -0,0 +1,491 @@
...
+                "Description": "Extracted from _FillValue metadata attribute"
+            }
+        ],
+        "MetadataSpecification": {
+            "URL": "https://cdn.earthdata.nasa.gov/umm/variable/v1.8.2",
+            "Name": "UMM-Var",
+            "Version": "1.8.2"
+        }
+    }
+]
```
Comment (eudoroolivares2016) [inline src/earthdataVarinfo/test/variables.json:491]: Add newline

File: src/earthdataVarinfo/test/expected_publish_response.json (hunk 4 above)
Comment (eudoroolivares2016) [inline src/earthdataVarinfo/test/expected_publish_response.json:9]: add newline

**Review by eudoroolivares2016 — APPROVED** [review]

**Pull request merged**